### ISSUE TITLE
Manage Organizations doesn't work

### ISSUE THREAD

**Issue by robsavoye** [hotosm/fmtm#417]
When I use the Manage Organizations button, the new organization doesn't get written to the database. Without an organization in the database, you can't go to the next page when creating a project.

**Comment by robsavoye** [comment]
When trying to create a new organization, I get an error on the logo. The log shows:
* [ERROR] fastapi | create_organization:74 | [Errno 13] Permission denied: 'app/images/sslogo2.jpg'. This prevents creating an organization from working. I have to manually add an org to the database to be able to create a project.

**Comment by spwoodcock** [comment]
## More info

## Adding the slug to the org

The error returned from the POST is:
```
{
	"errors": [
		{
			"loc": [
				"body",
				"slug"
			],
			"msg": "field required",
			"error": "field required['body', 'slug']"
		}
	]
}
```

The model for organization is:
```python
class Organisation(BaseModel):
    slug: str
    logo: str
    name: str
    description: str
    url: str
    type: int
```

**Solution**:
- We need to generate the slug param from the name, probably just a computed property with `name.toLowerCase()` will do for now.
- The slug should be included in the POSTed formData: https://github.com/hotosm/fmtm/blob/d11de09d4306e3dd2bc3a25b10d56870bc99d544/src/frontend/main/src/views/CreateOrganization.tsx#LL21C5-L21C13

### Image upload

- The issue around images that Rob posted is also a problem.
- We appear to be saving the uploaded logo image to /app/images/ on the API server.
- This isn't good practice, as restarting the server would result in losing the uploaded image.
tl;dr the image should be saved somewhere outside of the API server container to be persistent (and avoid file permission errors).

**Solution options**:
1. One option is to add the image to the database blob storage, then write an endpoint for getting images from the database. The URL provided in the organization model would be the endpoint url for getting the image.
2. Upload the image to an external object storage such as S3. This is the preferred option, but I'm not sure we have the infrastructure in place / this requires talks with IT to provision the S3.
3. Probably the most realistic: create a docker volume under /app/images that will persist on container restart (I can help with this).

**Comment by spwoodcock** [comment]
@nrjadkry it looks like you are already working on this and have solved the slug problem.

I can see that you handled the logo by adding the option to ignore it: https://github.com/hotosm/fmtm/blob/117d4e823fa21c6852d2a36ba0b79b9069d65813/src/backend/app/organization/organization_crud.py#L94

This would mean that orgs don't have logos, which works for now.

I will make a quick PR to add the docker volume mentioned above to help too 👍

**Comment by spwoodcock** [comment]
Fixed the permission issue with #514

**Comment by spwoodcock** [comment]
Note that the recent updates to the project endpoints/logic has broken the tests we have:
https://github.com/hotosm/fmtm/actions/runs/5344620635/jobs/9689213226

```
---------------------------- Captured log teardown -----------------------------
DEBUG    app.main:main.py:134 Shutting down FastAPI server.
=========================== short test summary info ============================
FAILED tests/test_projects.py::test_list_projects_summuries - assert 400 == 200
 +  where 400 = <Response [400]>.status_code
 +  and   200 = status.HTTP_200_OK
FAILED tests/test_projects.py::test_list_organisations - assert 1 == 3
 +  where 1 = len({'errors': [{'error': "value is not a valid integer['path', 'project_id']", 'loc': ['path', 'project_id'], 'msg': 'value is not a valid integer'}]})
 +    where {'errors': [{'error': "value is not a valid integer['path', 'project_id']", 'loc': ['path', 'project_id'], 'msg': 'value is not a valid integer'}]} = <bound method Response.json of <Response [400]>>()
 +      where <bound method Response.json of <Response [400]>> = <Response [400]>.json
FAILED tests/test_projects.py::test_list_projects_features - assert 400 == 200
 +  where 400 = <Response [400]>.status_code
 +  and   200 = status.HTTP_200_OK
FAILED tests/test_projects.py::test_project_logs - assert 400 == 200
 +  where 400 = <Response [400]>.status_code
 +  and   200 = status.HTTP_200_OK
FAILED tests/test_projects.py::test_categories - assert 404 == 200
 +  where 404 = <Response [404]>.status_code
 +  and   200 = status.HTTP_200_OK
FAILED tests/test_projects.py::test_get_task - assert 422 == 200
 +  where 422 = <Response [422]>.status_code
 +  and   200 = status.HTTP_200_OK
FAILED tests/test_projects.py::test_create_project - TypeError: unhashable type: 'dict'
FAILED tests/test_projects.py::test_delete_project - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_upload_project - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_upload_xlsform - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_upload_multi_polygon - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_upload_boundary - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_download_project - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_download_task - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_generate_files - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_create_organization - TypeError: unhashable type: 'dict'
FAILED tests/test_projects.py::test_preview_task - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_add_features - TypeError: Object of type set is not JSON serializable
18 failed, 3 passed in 3.28s
Error: Process completed with exit code 1.
```

**Comment by nrjadkry** [comment]
@spwoodcock where are we with the permission issue?

**Comment by spwoodcock** [comment]
Was it not fixed with my patch on development?

I'm really pushed for time recently and can't do much testing, sorry.

I'll be at FOSS4G all week starting today, so will perhaps have a bit of time - if you are getting errors, post a detailed error log here and I will try to fix 👍
